File: java/src/Leetcode1207.java
import java.util.*;

public class Leetcode1207 {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int num : arr){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        int valueSize = map.values().size();
        int setSize = new HashSet<>(map.values()).size();

        if(valueSize == setSize)
            return true;
        else
            return false;
    }
}

File: python/Leetcode1207.py
from typing import List

class Leetcode1207:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        map = {}
        for num in arr:
            if num in map.keys():
                map[num] = map[num] + 1
            else:
                map[num] = 1
        l = len(map.values())
        set_len = len(set(map.values()))        
        if set_len == l:
            return True
        else:
            return False
